service-workers/service-worker/resources/fetch-event-handled-worker.js

```diff
@@ -4,7 +4,8 @@
 // Send a message to the client with the client id.
 function send_message_to_client(message, clientId) {
   clients.get(clientId).then((client) => {
-    client.postMessage(message);
+    if (client)
+      client.postMessage(message);
```

Comment:
I see, client is null for the navigation? I'd prefer to handle this in the fetch event handler, as we should know it's the navigation there.

As a longer-term fix, it would also be neat to test event.handled for the navigation. We could do that using |reservedClientId| instead of |clientId| for the navigation. But making sure the message arrives at the test page at the right time may be tricky.

Reply:
Oh, I see, thanks for this suggestion.
Then I changed to use |resultingClientId| which seems to be the client id of simple.html when it's loaded. And the test is also updated to check the FetchEvent.handled for this navigation request in the new patch. Does the new change match your idea?